frontend/src/pages/signup.tsx

```diff
@@ -196,16 +216,23 @@ export default function SignUpPage() {
                   values.confirmPassword === ''
                     ? 'Required'
                     : values.password !== values.confirmPassword && values.submitFlag
-                    ? 'Passwords do not match'
-                    : ''
+                      ? 'Passwords do not match'
+                      : ''
                 }
                 value={values.confirmPassword}
               />
             </Grid>
           </Grid>
-          <Button type="submit" fullWidth variant="contained" sx={{ mt: 3, mb: 2 }}>
+          <Button id="signupBtn" type="submit" fullWidth variant="contained" sx={{ mt: 3, mb: 2 }}>
             Sign Up
```

Comment:
use condition rendering here, it should be:
` {loading ? <CircularProgress size={24} color="inherit" /> : 'Sign Up'}`